### PR TITLE
util: Add decrement of CQ ref counter in EP close

### PR DESCRIPTION
**Pull request by gladkovdmitry17**
+ Update MLX and RXM providers to use it

### CODE REVIEW

**Comment by gladkovdmitry17** [comment]
@a-ilango @shefty Could you review, please? Thanks in advance.
It was missed in the PR #3103

File: prov/util/src/util_ep.c [hunk 1]
```diff
@@ -129,6 +129,21 @@ int ofi_endpoint_init(struct fid_domain *domain, const struct util_prov *util_pr
 int ofi_endpoint_close(struct util_ep *util_ep)
 {
 	fastlock_destroy(&util_ep->lock);
+
+	if (util_ep->tx_cq) {
+		ofi_atomic_dec32(&util_ep->tx_cq->ref);
```
Comment (a-ilango) [inline prov/util/src/util_ep.c]: This should come after fid_list_remove. Only after that we are not using the cq anymore.

File: prov/util/src/util_ep.c [hunk 2]
```diff
@@ -129,6 +129,21 @@ int ofi_endpoint_init(struct fid_domain *domain, const struct util_prov *util_pr
...
+		fid_list_remove(&util_ep->tx_cq->ep_list,
+				&util_ep->tx_cq->ep_list_lock,
+				&util_ep->ep_fid.fid);
+	}
+
+	if (util_ep->rx_cq) {
+		ofi_atomic_dec32(&util_ep->rx_cq->ref);
```
Comment (a-ilango) [inline prov/util/src/util_ep.c]: Ditto.

File: prov/util/src/util_ep.c (hunk 1 above)
Comment (gladkovdmitry17) [inline prov/util/src/util_ep.c]: Ok, will do that

**Pull request merged**